.github/workflows/push-mimir-build-image.yml

```diff
@@ -0,0 +1,141 @@
+        run: |
+          # The case when workflow is triggered by pull request
+          if [[ ${{ github.event_name }} == 'pull_request' && ${{ github.event.pull_request.user.login }} == 'dependabot[bot]' ]]; then
+            echo "triggered=true" >> $GITHUB_OUTPUT
+            echo "pull_request_number=${{ github.event.pull_request.number }}" >> $GITHUB_OUTPUT
+            echo "pull_request_url=${{ github.event.pull_request.html_url }}" >> $GITHUB_OUTPUT
+            echo "login=${{ github.event.pull_request.user.login }}" >> $GITHUB_OUTPUT
+          # The case when workflow is triggered by comment on pull request and it matches the trigger
+          elif [[ ${{ github.event_name }} == 'issue_comment' && ${{ steps.trigger.outputs.triggered }} == 'true' ]]; then
+            echo "triggered=true" >> $GITHUB_OUTPUT
+            echo "pull_request_number=${{ github.event.issue.number }}" >> $GITHUB_OUTPUT
+            # echo "pull_request_url=${{ github.event.issue.html_url }}" >> $GITHUB_OUTPUT
+            echo "login=${{ github.event.issue.author }}" >> $GITHUB_OUTPUT
+          else
+            echo "triggered=false" >> $GITHUB_OUTPUT
+          fi
+    outputs:
+      triggered: ${{ steps.variables.outputs.triggered }}
+      pull_request_number: ${{ steps.variables.outputs.pull_request_number }}
+      pull_request_url: ${{ steps.variables.outputs.pull_request_url }}
+      login: ${{ steps.variables.outputs.login }}
+
+  build_and_push:
+    runs-on: ubuntu-latest
+    needs: prepare_variables
+    if: needs.prepare_variables.outputs.triggered == 'true'
+    # If the PR is raised by dependabot, then build the image
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Checkout Pull Request Branch
+        run: gh pr checkout ${{ needs.prepare_variables.outputs.pull_request_number }}
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Prepare Variables
+        id: prepare
+        run: |
+          echo "path=mimir-build-image/Dockerfile" >> $GITHUB_OUTPUT
+          echo "image=grafana/mimir-build-image" >> $GITHUB_OUTPUT
+
+      - name: Compute Image Tag
+        id: compute_hash
+        run: |
+          current_hash=$(md5sum ${{ steps.prepare.outputs.path }} | awk '{print $1}')
+          echo "the file path is ${{ steps.prepare.outputs.path }}"
+          echo "build tag is $current_hash"
+          tag="${{ needs.prepare_variables.outputs.pull_request_number }}_$current_hash"
+          echo "tag=$tag" >> $GITHUB_OUTPUT
+
+      - name: Check Should Build Image
+        id: check_build
+        run: |
```

Comment:
Should we use a command-line tool like [`skopeo`](https://github.com/containers/skopeo) to check if the image exists? We use this in other (non-public) repositories for this exact purpose, and it supports authentication when talking to the Docker Hub API, which would help avoid rate limiting as well.

Reply:
yes, you are right, Peter raised same concerns, I plan to change it before putting it to ready to review.